src/devices/Common/System/Device/Spi/SpiBusInfo.cs

```diff
@@ -1,19 +1,41 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.IO;
+using System.Runtime.InteropServices;
+
 namespace System.Device.Spi
 {
+    /// <summary>
+    /// Base class for SPI bus information.
+    /// </summary>
     public static class SpiBusInfo
     {
         private static int s_bufferSize = GetBufferSize();
 
-        // we would need to decide if this should throw when size can't be found or just return -1 or other placeholder
+        /// <summary>
+        /// Buffer size assigned to the SPI bus or -1 if the buffer size is not available.
+        /// </summary>
         public static int BufferSize => s_bufferSize;
 
         private static int GetBufferSize()
         {
-            // non-throwing version of int.Parse(File.ReadAllText("/sys/module/spidev/parameters/bufsiz"))
-            // return -1 on error
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                const string parameterPath = "/sys/module/spidev/parameters/bufsiz";
+                if (File.Exists(parameterPath) && int.TryParse(File.ReadAllText(parameterPath), out int buferSize))
+                {
+                    return buferSize;
+                }
+                else
+                {
+                    return -1;
+                }
+            }
+            else
+            {
+                throw new PlatformNotSupportedException("This value is only supported on Linux operating systems");
```

Comment:
This should not throw, never. This would now throw a TypeInitalizationException when just referencing this type on Windows. This is especially critical since this code executes in a static constructor.

Reply:
yes, if we return -1 when info is not available let's stick to that